BinaryTreeTest.java

```diff
@@ -0,0 +1,71 @@
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+/**
+ * Created by adi on 6/1/2017.
+ */
+public class BinaryTreeTest {
+
+    @Test
+    public void testPrintAncestors(){
+        BinaryTree<Integer> myTree = new BinaryTree(16);
+        myTree.setInsertChild(BinaryTree.InsertChild.Left);
+        myTree.insert(9, 16);
+        myTree.insert(3, 9);
+        myTree.insert(1, 3);
+        myTree.setInsertChild(BinaryTree.InsertChild.Right);
+        myTree.insert(18, 16);
+        myTree.insert(14, 9);
+        myTree.insert(5, 3);
+        myTree.insert(19, 18);
+
+        List<Integer> result = new ArrayList<>();
+        result.add(3);
+        result.add(9);
+        result.add(16);
+
+        assertEquals(result , myTree.printAncestors(myTree, 5));
+
+        assertNull(myTree.printAncestors(myTree, 4));
+
+        result.remove(2);
+        result.remove(1);
+        result.remove(0);
+
+        assertEquals(result, myTree.printAncestors(myTree, 16));

```

Comment:
Typically each test method should verify only one scenario and have name that is informative about the case (and expected result if possible - it not always makes sense though). Could you split test methods to single-scenario ones?
For example testPrintAncestors_existingNode and testPrintAncestors_missingNode or something similar.

The same applies to tests for lowestCommonAncestor.